librz/core/cmd/cmd_debug.c

```diff
@@ -4410,23 +4410,67 @@ RZ_IPI RzCmdStatus rz_cmd_debug_add_bp_handler(RzCore *core, int argc, const cha
 
 // dbl
 RZ_IPI RzCmdStatus rz_cmd_debug_list_bp_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
-	RzOutputMode mode = state->mode;
-	switch (mode) {
-	case RZ_OUTPUT_MODE_STANDARD:
-		rz_bp_list(core->dbg->bp, 0);
-		break;
-	case RZ_OUTPUT_MODE_RIZIN:
-		rz_bp_list(core->dbg->bp, 1);
-		break;
-	case RZ_OUTPUT_MODE_JSON:
-		rz_bp_list(core->dbg->bp, 'j');
-		break;
-	case RZ_OUTPUT_MODE_QUIET:
-		rz_bp_list(core->dbg->bp, -1);
-		break;
-	default:
-		rz_warn_if_reached();
-		break;
+	RzBreakpointItem *b;
+	RzListIter *iter;
+	RzOutputMode mode = RZ_OUTPUT_MODE_STANDARD;
+	PJ *pj = NULL;
+	if (state) {
+		mode = state->mode;
+		pj = state->d.pj;
+		rz_cmd_state_output_array_start(state);
+	}
+	rz_list_foreach (core->dbg->bp->bps, iter, b) {
+		switch (mode) {
+		case RZ_OUTPUT_MODE_STANDARD: {
+			rz_cons_printf("0x%08" PFMT64x " - 0x%08" PFMT64x
+						 " %d %c%c%c %s %s %s %s cmd=\"%s\" cond=\"%s\" "
+						 "name=\"%s\" module=\"%s\"\n",
+				b->addr, b->addr + b->size, b->size,
+				((b->perm & RZ_BP_PROT_READ) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'r' : '-',
+				((b->perm & RZ_BP_PROT_WRITE) | (b->perm & RZ_BP_PROT_ACCESS)) ? 'w' : '-',
+				(b->perm & RZ_BP_PROT_EXEC) ? 'x' : '-',
+				b->hw ? "hw" : "sw",
+				b->trace ? "trace" : "break",
+				b->enabled ? "enabled" : "disabled",
+				rz_bp_is_valid(core->dbg->bp, b) ? "valid" : "invalid",
+				rz_str_get(b->data),
+				rz_str_get(b->cond),
+				rz_str_get(b->name),
+				rz_str_get(b->module_name));
+			break;
+		}
+		case RZ_OUTPUT_MODE_RIZIN: {
+			if (b->module_name) {
+				core->dbg->bp->cb_printf("dbm %s %" PFMT64d "\n", b->module_name, b->module_delta);
+			} else {
+				core->dbg->bp->cb_printf("db @ 0x%08" PFMT64x "\n", b->addr);
+			}
+			break;
+		}
+		case RZ_OUTPUT_MODE_JSON: {
+			pj_o(pj);
+			pj_kN(pj, "addr", b->addr);
+			pj_ki(pj, "size", b->size);
+			pj_ks(pj, "perm", rz_str_rwx_i(b->perm & 7)); /* filter out R_BP_PROT_ACCESS */
```

Comment:
why here you use the rz_str_rwx_i but not above? are we also sure that the enums are mapped in the same way? also why filtering out the access flag? this looks wrong

Reply:
This was the way the old command was implemented. I will change it.

Reply:
yes, please.

Reply:
I have changed it. Removed the deprecated `RZ_BP_PROT` enum.